src/server/dflycmd.cc

```diff
@@ -379,6 +379,7 @@ void DflyCmd::TakeOver(CmdArgList args, ConnectionContext* cntx) {
 
   // TODO: We should cancel blocking commands before awaiting all
   // dispatches to finish.
+  sf_->CancelBlockingCommands();
```

Comment:
can you please add a comments on why we call AwaitDispatches, why we must call CancelBlockingCommands and why we need the TransactionGuard

Reply:
also why we disable_expirations